crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -27,6 +29,13 @@ pub trait C8YHttpProxy: Send + Sync {
 
     async fn get_jwt_token(&mut self) -> Result<SmartRestJwtResponse, SMCumulocityMapperError>;
```

Comment:
This can be done but `automock` won't work with trait extensions (probably another reason to stop using `automock`). Will address this in an upcoming PR by using `mock!` macro to mock these traits.